csharp/src/Ice/BZip2.cs

```diff
@@ -48,321 +48,78 @@ public BZStream(IntPtr nextOut, uint availOut)
-                                                             int workFactor);
-
-        [DllImport("libbz2.dylib", EntryPoint = "BZ2_bzCompress", ExactSpelling = true)]
-        internal static extern int MacOSBZ2_bzCompress(ref BZStream stream, int action);
-
-        [DllImport("libbz2.dylib", EntryPoint = "BZ2_bzCompressEnd", ExactSpelling = true)]
-        internal static extern int MacOSBZ2_bzCompressEnd(ref BZStream stream);
-
-        [DllImport("libbz2.dylib", EntryPoint = "BZ2_bzBuffToBuffDecompress", ExactSpelling = true)]
-        internal static extern int MacOSBZ2_bzBuffToBuffDecompress(byte[] dest,
-                                                                   ref int destLen,
-                                                                   byte[] source,
-                                                                   int sourceLen,
-                                                                   int small,
-                                                                   int verbosity);
-    }
-
-    internal delegate int CompressBuffer(byte[] dest,
-                                         ref int destLen,
-                                         byte[] source,
-                                         int sourceLen,
-                                         int blockSize100k,
-                                         int verbosity,
-                                         int workFactor);
-
-    internal delegate int DecompressBuffer(byte[] dest,
-                                           ref int destLen,
-                                           byte[] source,
-                                           int sourceLen,
-                                           int small,
-                                           int verbosity);
-
-    internal delegate int BZCompressInit(ref BZStream stream, int blockSize100k, int verbosity, int workFactor);
-    internal delegate int BZCompress(ref BZStream stream, int action);
-    internal delegate int BZCompressEnd(ref BZStream stream);
-
     internal class BZip2
     {
-        private const int BzRun = 0;
-        private const int BzFinish = 2;
+        private enum BzAction
+        {
+            Run = 0,
+            Flush = 1,
+            Finish = 2
+        }
 
-        private const int BzRunOk = 1;
-        private const int BzFinishOk = 3;
-        private const int BzStreamEnd = 4;
+        private enum BzStatus
+        {
+            RunOk = 1,
+            FinishOk = 3,
+            StreamEnd = 4,
+
+            SequenceError = -1,
+            ParamError = -2,
+            MemError = -3,
+            DataError = -4,
+            DataErrorMagic = -5,
+            IoError = -6,
+            UnexpectedEof = -7,
+            OutbuffFull = -8,
+            ConfigError = -9
+        }
 
+        internal static string LibName { get; private set; }
```

Comment:
I would make this LibName internal. We don't need it in error messages from other classes.

Reply:
do you mean private?

Reply:
Yes, private!